spec/PeriodicBackgroundSync-index.bs

```diff
@@ -110,3 +112,97 @@ The user agent should cap the duration and frequency of these events to limit re
 Large resources should be downloaded by registering a [=background fetch=] via the {{BackgroundFetchManager}} interface.
 
 In addition, the user agent should consider other factors such as user engagement with the origin, and any user indications to temporarily reduce data consumption, such as a Data Saving Mode, to adjust the frequency of <code>periodicsync</code> events.
+
+Algorithms {#algorithms}
+=========================
+## <dfn>Calculate a time to fire</dfn> ## {#caculate-time-to-fire}
+This section describes how a user agent can calculate the [=time to fire=] for a [=periodicsync registration=], |registration|. The time interval between periodicSync events for |registration| MUST be greater than or equal to the {{options/minInterval}} value associated with |registration|.
+
+The user agent MAY include factors such as user engagement with the origin to decide this time interval, allowing origins with high user engagement to update their web apps more often. In addition, the user agent SHOULD have per-origin and cross-origin caps on this interval, as recommended in [[#privacy]] and [[#resources]] sections.
+
+The user agent may also decide to retry each failed periodicSync event, capped to a user agent defined <dfn>maximum number of retries</dfn>.
```

Comment:
I'd add a type here, the algorithm below would also benefit from types for variables

Reply:
done.